### PR TITLE
Fix #160 - support for ReduceLROnPlateau

### LINKED ISSUE #160: Loss dependent scheduler not supported (e.g. ReduceLROnPlateau)
**Describe the bug**

When using ReduceLROnPlateau scheduler, the scheduler expects input of epoch loss value to identify Plateau and raises error.
Adding stopping_loss as parameter to scheduler.step() in the code also does not help since it is located in the training section 
and not after the validation evaluation section.
I recommend as solution to move the scheduler.step lines to the Class TabModel section in the def of fit (line 244) and include stopping_loss (this time it is valid stopping_loss) as parameter.

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**Expected behavior**

**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

### PR DESCRIPTION
**Pull request by bennyjg**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

**What kind of change does this PR introduce?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

**Closing issues**
Closes #160
- Benny
Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pytorch_tabnet/tab_model.py [hunk 1]
```diff
@@ -242,6 +242,12 @@ def fit(self, X_train, y_train, X_valid=None, y_valid=None, loss_fn=None,
             else:
                 self.patience_counter += 1
 
+            if self.scheduler is not None:
+				if self.scheduler_fn == torch.optim.lr_scheduler.ReduceLROnPlateau:
```
Comment (Optimox) [inline pytorch_tabnet/tab_model.py:246]: You have an indentation problem here I think.

More over, I think this a bit too specific for ReduceLROnPlateau... but at least could you make the if statement use isinstance instead of == ?

**Review by Optimox — CHANGES_REQUESTED** [review]
You'll also need to make only one commit and change the name to something like `feat: enable LRscheduler OnPlateau`

**Comment by bennyjg** [comment]
Replaced by PR #179
